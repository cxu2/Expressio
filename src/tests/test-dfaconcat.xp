main : -> int {
int ans;
dfa zz;
dfa oo;
dfa u;
zz  = {
 states : 3
 alphabet : ['0', '1']
 start : 0
 final : [2]
 transitions : [ (0 '0' 1), (1 '0' 2)] 
};

oo = {
 states : 3
 alphabet : ['0', '1']
 start : 0
 final : [2]
 transitions : [ (0 '1' 1), (1 '1' 2) ]
};

printdfa(zz);
printdfa(oo);
u = zz concat oo;

printdfa(u);

if(u accepts "00"){
  printf("u accepts 00");
}  else {
   printf("u rejects 00");
}

if(u accepts "0011"){
  printf("u accepts 0011");
}  else	    {
   printf("u rejects 0011");
}


if(u accepts "11"){
  printf("u accepts 11");
}  else	    {
   printf("u rejects 11");
}

return 0;
}
