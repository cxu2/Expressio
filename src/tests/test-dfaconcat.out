
nstates:  3
alphabet: 0 1 
nsym:     2
start:    0
fin:      2 
nfin:     1
delta:    
        |   0 |   1 
    ----|-----|-----
       0|   1    -1  
       1|   2    -1  
       2|  -1    -1  

nstates:  3
alphabet: 0 1 
nsym:     2
start:    0
fin:      2 
nfin:     1
delta:    
        |   0 |   1 
    ----|-----|-----
       0|  -1     1  
       1|  -1     2  
       2|  -1    -1  

nstates:  5
alphabet: 0 1 
nsym:     2
start:    0
fin:      4 
nfin:     1
delta:    
        |   0 |   1 
    ----|-----|-----
       0|   1    -1  
       1|   2    -1  
       2|  -1     3  
       3|  -1     4  
       4|  -1    -1  
u rejects 00
u accepts 0011
u rejects 11
